gatsby-node.js

```diff
@@ -1,31 +1,43 @@
 const { createRedirects } = require('./gatsby/create-pages/create-redirects');
 const {
-  createPreviewCard,
-} = require('./gatsby/create-node/create-preview-card');
+  createPreviewCards,
+} = require('./gatsby/create-node/create-preview-cards');
 const { createBlogPosts } = require('./gatsby/create-pages/create-blog-posts');
 const {
-  createCareerDetails,
-} = require('./gatsby/create-pages/create-career-details');
+  createCareerPages,
+} = require('./gatsby/create-pages/create-career-pages');
 const { createFilePath } = require('gatsby-source-filesystem');
+const {
+  createCareerPagePathField,
+} = require('./gatsby/create-node/create-career-page-path');
 
 exports.onCreateNode = (gatsbyCreateNodeArgs) => {
-  createPreviewCard(gatsbyCreateNodeArgs);
+  createPreviewCards(gatsbyCreateNodeArgs);
+
   const { node, actions, getNode } = gatsbyCreateNodeArgs;
   const { createNodeField } = actions;
-
+  /**
+   * Provide a slug for any markdown page (which is not given by default)
+   * to build our pages based on this markdown file.
+   */
   if (node.internal.type === `MarkdownRemark`) {
     const value = createFilePath({ node, getNode });
     createNodeField({
-      name: `slug`,
+      name: `path`,
       node,
       value,
     });
   }
+
+  if (node.internal.type === `SyPersonioJob`) {
+    createCareerPagePathField({ node, createNodeField });
```

Comment:
Instead of adding a `path` to the personio jobs through source we let the source plugin not know about this issue and instead create the required path field here. Only different: It's provided in `{fields: {path}}` instead of `{path}` which caused some refactoring when I changed it.